arviz/plots/backends/bokeh/autocorrplot.py

```diff
@@ -57,12 +59,16 @@ def plot_autocorr(
         title = Title()
         title.text = make_label(var_name, selection)
         ax.title = title
+        ax.x_range = DataRange1d(
+            start=0, end=max_lag, bounds=(0, len(np.arange(len(y)))), min_interval=5
+        )
+        ax.y_range = DataRange1d(start=-1, end=1, bounds=rcParams["plot.bokeh.bounds"])
 
-    if axes.size > 0:
-        axes[0, 0].x_range._property_values["start"] = 0  # pylint: disable=protected-access
-        axes[0, 0].x_range._property_values["end"] = max_lag  # pylint: disable=protected-access
-        axes[0, 0].y_range._property_values["start"] = -1  # pylint: disable=protected-access
-        axes[0, 0].y_range._property_values["end"] = 1  # pylint: disable=protected-access
+    for _, ax in zip(
+        plotters, (item for item in axes.flatten() if item is not None)
+    ):
+        ax.x_range = axes[0, 0].x_range
+        ax.y_range = axes[0, 0].y_range
```

Comment:
I think in [this](https://arviz-devs.github.io/arviz/examples/bokeh/bokeh_plot_autocorr.html) case, ```len(y)``` will return 500 but it has values from 0 to 499 only right?



> I think this is not necessary now, the range has already been set above.

I did this to ensure link panning. Without this, the plots were not panning all together like they used to. Should I remove this?

Reply:
Yes, values will go between 0 and 499, what I meant is that `len(np.arange(len(y))))` should be equal to `len(y)`

Reply:
Regarding panning, I am not sure I understand the issue, I am not extremely familiar with bokeh yet. One difference between adding the extra loop or not is that with extra loop, all plots use the same `DataRange1d` object, whereas with only the first loop, each uses a different `DataRange1d` object with the same parameters. Could this make a difference?

Reply:
> Regarding panning, I am not sure I understand the issue, I am not extremely familiar with bokeh yet. One difference between adding the extra loop or not is that with extra loop, all plots use the same `DataRange1d` object, whereas with only the first loop, each uses a different `DataRange1d` object with the same parameters. Could this make a difference?

I tested it with [this](https://arviz-devs.github.io/arviz/examples/bokeh/bokeh_plot_autocorr.html) example code (which already supports linked panning i.e. panning any one plot will pan all others in the same way) and didn't find any issues. I looked up the solution for linked panning and [bokeh docs](https://docs.bokeh.org/en/latest/docs/user_guide/interaction/linking.html) suggested to provide the range of the first plot as the range to other plots. That's why I though this might be the solution.

What do you think? @OriolAbril @ahartikainen?

Reply:
That sounds reasonable. If possible, I would like to avoid two loops over the same set of axes. 

Does defining a `data_range_x` and `data_range_y` **before** the loop and using always this same object (as opposed to creating a new one every iteration as it is currently done) solve the issue?

Reply:
I thought of that too but ```x_range``` needs ```len(y)``` and ```y``` is present in the loop.

Maybe inside the first loop we can check if the ```axes``` is the first one i.e. the first plot and assign the ranges for that plot only and for every other plot we assign the range equal to the range of the first plot.

For example:
```
for loop:
    -----
    -----
    if ax is first:
         axes[0, 0].x_range = DataRange1d(
            start=0, end=max_lag, bounds=(0, len(np.arange(len(y)))), min_interval=5
        )
         axes[0, 0].y_range = DataRange1d(start=-1, end=1, bounds=rcParams["plot.bokeh.bounds_y_range"])
    else:
         ax.x_range = axes[0, 0].x_range
         ax.y_range = axes[0, 0].y_range
```

What do you think?

Reply:
I think `y` should have length `n_draws` or `n_draws * n_chains` depending on the value of `combine`.

Reply:
Sorry now I got your point! Yes, this will solve the panning issue. How do I access the ```n_draws``` and ```n_chains``` values?

Reply:
Do I need to pass the data from ```plots/autocorrplot.py``` to ```plots/backends/bokeh/autocorrplot.py``` i.e. from frontend to backend to access the ```data["draw"]``` and ```data["chain"]```?

Reply:
something like:

    chain, draw, *_ = plotters[0][2].shape

should work

Reply:
This gives the following error:

```ValueError: not enough values to unpack (expected at least 2, got 1)```

Should we check in single dimension only like ```plotters[0].shape```?

Reply:
Maybe we just go with two loop solution?

No need to create extra code and num of axes is usually (always) low.

Reply:
I gave a quick look to the code creating plotters, and the combination of chains is handled there. Therefore `len(plotters[0][2])` is equal to `len(y)`, and it will always be a 1d array. The flatten on `x` there there

Reply:
So should I remove the ```combined``` argument from both the backends too?

Reply:
My bad, the combined boolean is still needed, I mistook the helper function used by another. If we call `x = plotters[0][2]` there are 2 cases: 
* combined=True -> x.shape = (chain, draw) -> flatten is needed
* combined=False -> x.shape = (draw) -> no flatten
However, as the data used is always the flattened version, what is true is `len(y) = x.size`

Reply:
I see. That's why using length as ```plotters[0][2]``` worked for me

Reply:
It should be `.size` otherwise it won't give the correct value with combined=True

Reply:
And do I need to flatten it as well for ```combined=True``` or is it already flattened because I'm calling ```len_y = plotters[0][2].size``` **before** the loop

Reply:
[`.size`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.size.html) only returns an integer, `x_prime` still has to be flattened if combined=True

Reply:
Got it. I don't know why it slipped from my mind :)